### DISCUSSION TITLE
Table action does not return a file to download

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#7001] [Help] [answered]
I have a table action that creates a pdf and returns it to the brwoser to display.

```
Tables\Actions\Action::make('print_pdf')
	->label('PDF')
	// this DOES work as the controller returns a response
	->url(fn (Invoice $record): string => route('invoice.pdf', ['invoice' => $record->id]))
	
	// this does NOT work. The files is generated but not returned to the browser
	->action(function ($record) {
		$action = app(CreateInvoicePdfAction::class);
		$file = $action->handle($record);

		return new Response(file_get_contents($file), 200, array(
			'Content-Type' => 'application/pdf',
			'Content-Disposition' => 'attachment; filename="' . basename($file) . '"'
		));
	})
	->iconButton()
	->color('success')
```

As noted inside the code I have to use a controller using the url function for the generated file to be returned to the browser. 
When I try to use the action nothing is returned. 

The controller function looks like this and works as expected:

```
public function pdf(Invoice $invoice): Response
{
	$file = $this->_createInvoicePdfAction->handle($invoice);
	$filename = basename($file);

	return new Response(file_get_contents($file), 200, array(
		'Content-Type' => 'application/pdf',
		'Content-Disposition' => 'attachment; filename="' . $filename . '"'
	));
}
```

I can live with using a controller in a table action. But now I wanted to create a bulk action to render multiple pdf files. Unfortunately a bulk action does not have a url method. I am stuck with the action method.

Is there any way to fix this? 

Thank you

**Comment by danharrin** [comment, marked as answer]
`return response()->download()` https://laravel-livewire.com/docs/2.x/file-downloads

**Reply by OvalMedia** [reply]
Thx
